sklearn/preprocessing/_polynomial.py

```diff
@@ -1150,9 +1220,6 @@ def transform(self, X):
                     " transformer to produce fewer than 2^31 output features"
                 )
             XBS = sparse.hstack(output_list, format="csr")
-        elif self.sparse_output:
-            # TODO: Remove ones scipy 1.10 is the minimum version. See comments above.
-            XBS = sparse.csr_matrix(XBS)
```

Comment:
Why remove those lines. I find it safer to always ensure we convert back to CSR whatever the lil conversion logic that happens before (and that depends on many parameters).

Reply:
My confusion stems from the fact that the moved lines now show above (at line 1091) instead of below in the current state of the aggregate diff of the PR at the time of writing.

But I guess moving those lines results in more logically organized code, hopefully. Anyway this code will become much simpler when we drop support scipy < 1.10 so this is no big deal as long at the existing tests still pass.

Reply:
Now I have seen that moving those lines here led to CI failures for `test_spline_transformer_handles_all_nans[True-periodic]` and `[True-continue]` for scipy versions 1.8 and 1.9. I now have moved it back to where it was.